tests/Http/Productions/ProductionsWriteControllerTest.php

```diff
@@ -59,14 +59,13 @@ public function it_can_create_production(): void
             ]
         );
 
-        $expectedEvents = [$eventId1, $eventId2];
-        $this->commandBus->expects($this->once())->method('dispatch')->willReturnCallback(
-            function (GroupEventsAsProduction $command) use ($name, $expectedEvents) {
-                $this->assertEquals($name, $command->getName());
-                $this->assertEquals($expectedEvents, $command->getEventIds());
-            }
-        );
+        $this->commandBus->record();
         $this->controller->create($request);
+
+        $recordedCommand = $this->commandBus->getRecordedCommands()[0];
+        $this->assertInstanceOf(GroupEventsAsProduction::class, $recordedCommand);
+        $this->assertEquals($name, $recordedCommand->getName());
+        $this->assertEquals([$eventId1, $eventId2], $recordedCommand->getEventIds());
```

Comment:
Nit: Should've mentioned it in the previous review, but I was always taught it was better to re-create the object you'd expect and then compare it using `assertEquals()` whenever possible, without getters. So you're effectively comparing the inner state.

Also if you do it like:
```
$expectedCommands = [new GroupEventsAsProduction(...)]; // Note the array
$this->assertEquals($expectedCommands, $recordedCommands);
```
You can verify that no extra commands were dispatched.

But it's definitely nitpicking and good enough as-is, so I'm approving either way.

Reply:
As a compromise, you could also assert the count of the `$recordedCommands` being `1` of course. I think the biggest advantage of building an `$expected` array/object is just readability and less cognitive load.